### PR TITLE
Handle nested objects and arrays

### PR DESCRIPTION
**Pull request by dhaneshnm**
Check for object match with in an array.

### CODE REVIEW

File: src/shared/utils/index.ts [hunk 1]
```diff
@@ -357,10 +357,20 @@ export const matchPlugins = (
...
     if (Array.isArray(other) && !Array.isArray(value)) {
-      return other.length === 1 && other.indexOf(value) >= 0;
+      return other.length === 1 && isMatch(value, other[0]);
     }
+
+    // return true if any object in value array match any object in
+    // other array.
     if (Array.isArray(other) && Array.isArray(value)) {
-      return difference(other, value).length === 0;
+      for (let i = 0; i < value.length; i += 1) {
```
Comment (Stafeeva) [inline src/shared/utils/index.ts]: it can be just i++ :) classic!

File: src/shared/utils/__tests__/index.spec.ts [hunk 2]
```diff
@@ -301,5 +302,65 @@ describe('utils functions', () => {
       };
       expect(matchPlugins(pluginsMap, plugins, resource)).toEqual([]);
     });
+
+    it('matches a resource with multiple plugins', () => {
+      const plugins: string[] = ['morphology', 'morphology2'];
+      const pluginsMap = {
+        morphology: {
+          // '@type': ['ReconstructedCell'],
```
Comment (Stafeeva) [inline src/shared/utils/__tests__/index.spec.ts]: is this comment needed here?

File: src/shared/utils/index.ts (hunk 1 above)
Comment (dhaneshnm) [reply src/shared/utils/index.ts]: Tslint is complaining about ++. https://palantir.github.io/tslint/rules/increment-decrement/
I prefer ++ and -- to be honest.

File: src/shared/utils/__tests__/index.spec.ts (hunk 2 above)
Comment (dhaneshnm) [reply src/shared/utils/__tests__/index.spec.ts]: Nope. I have removed it.

File: src/shared/utils/index.ts (hunk 1 above)
Comment (Stafeeva) [reply src/shared/utils/index.ts]: fair enough :)

**Pull request merged**